nav2_route/include/nav2_route/interfaces/graph_file_loader.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2023 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_ROUTE__INTERFACES__GRAPH_FILE_LOADER_HPP_
+#define NAV2_ROUTE__INTERFACES__GRAPH_FILE_LOADER_HPP_
+
+#include <string>
+#include <memory>
+
+#include "nav2_route/types.hpp"
+
+namespace nav2_route
+{
+
+/**
+ * @class GraphFileLoader
+ * @brief A plugin interface to parse a file into the graph
+ */
+class GraphFileLoader
+{
+public:
+  using Ptr = std::shared_ptr<GraphFileLoader>;
+
+  /**
+   * @brief Constructor
+   */
+  GraphFileLoader() = default;
+
+  /**
+   * @brief Virtual destructor
+   */
+  virtual ~GraphFileLoader() = default;
+
+  /**
```

Comment:
This should have a validate file API to check if this file exists and is valid. Different specifications might have different ways of wanting to do that (e.g. xml can be verified via STL, as we do in BT Navigator before attempting to load a file to know if its not valid)